main.cpp

```diff
@@ -112,325 +112,382 @@ struct DoubleType;
-    FloatType &multiply(float num);
-    FloatType &divide(float num);
+    FloatType& add(float num);
+    FloatType& subtract(float num);
+    FloatType& multiply(float num);
+    FloatType& divide(float num);
 
-    FloatType &add(const IntType& it);
-    FloatType &subtract(const IntType& it);
-    FloatType &multiply(const IntType& it);
-    FloatType &divide(const IntType& it);
+    FloatType& add(const IntType& it);
+    FloatType& subtract(const IntType& it);
+    FloatType& multiply(const IntType& it);
+    FloatType& divide(const IntType& it);
 
-    FloatType &add(const DoubleType& dt);
-    FloatType &subtract(const DoubleType& dt);
-    FloatType &multiply(const DoubleType& dt);
-    FloatType &divide(const DoubleType& dt);
+    FloatType& add(const FloatType& it);
+    FloatType& subtract(const FloatType& it);
+    FloatType& multiply(const FloatType& it);
+    FloatType& divide(const FloatType& it);
+
+    FloatType& add(const DoubleType& dt);
+    FloatType& subtract(const DoubleType& dt);
+    FloatType& multiply(const DoubleType& dt);
+    FloatType& divide(const DoubleType& dt);
 
-    float* value = nullptr;
     FloatType(float input): value(new float(input)){}
     ~FloatType(){ delete value;}
+    
+    float* value = nullptr;
 };
 
-IntType &IntType::add(int num)
+IntType& IntType::add(int num)
 {
-    *(this->value) += num;
-
+    *value += num;
     return *this;
 }
-IntType &IntType::subtract(int num)
+IntType& IntType::subtract(int num)
 {
-    *(this->value) -= num;
-
+    *value -= num;
     return *this;
 }
-IntType &IntType::multiply(int num)
+IntType& IntType::multiply(int num)
 {
-    *(this->value) *= num;
-    
+    *value *= num;
     return *this;
 }
 
-IntType &IntType::divide(int num)
+IntType& IntType::divide(int num)
 {
     if(num == 0)
     {
         std::cout << "error: integer division by zero is an error and will crash the program!"<< std::endl;
         return *this;
     }
 
-    *(this->value) /= num;
+    *value /= num;
     return *this;
 }
 
-IntType &IntType::add(const FloatType& ft)
+IntType& IntType::add(const IntType& it)
 {
-    *(this->value) += *(ft.value);
-    return *this;
+    return add(*it.value);
 }
-IntType &IntType::subtract(const FloatType& ft)
+IntType& IntType::subtract(const IntType& it)
 {
-    *(this->value) -= *(ft.value);
-    return *this;
+    return subtract(*it.value);
 }
-IntType &IntType::multiply(const FloatType& ft)
+IntType& IntType::multiply(const IntType& it)
 {
-    *(this->value) *= *(ft.value);
-    return *this;
+    return multiply(*it.value);
 }
-IntType &IntType::divide(const FloatType& ft)
+IntType& IntType::divide(const IntType& it)
 {
-    if(*(ft.value) == 0)
+    if(*it.value == 0)
     {
         std::cout << "error: integer division by zero is an error and will crash the program!"<< std::endl;
         return *this;
     }
-    *(this->value) /= *(ft.value);
-    return *this;
+    
+    return divide(*it.value);
```

Comment:
your `IntType::divide(int)` already has this std::cout message in it.
You don't need to duplicate the message here
all you need to do is call `return divide(*it.value);` in this function.
- [ ] revise